docs/ReferenceGuides/UnderscoredAttributes.md

```diff
@@ -0,0 +1,328 @@
+
+Allows access to `self` inside a closure without explicitly capturing it,
+even when `Self` is a reference type.
+
+```swift
+class C {
+  func f() {}
+  func g(_: @escaping () -> Void {
+    g({ f() }) // error: call to method 'f' in closure requires explicit use of 'self'
+  }
+  func h(@_implicitSelfCapture _: @escaping () -> Void) {
+    h({ f() }) // ok
+  }
+}
+```
+
+## `@_inheritActorContext`
+
+(Note that it is "inherit", not "inherits", unlike below.)
+
+## `@_inheritsConvenienceInitializers`
+
+## `@_marker`
+
+Indicates that a protocol is a marker protocol;
+one that represents some meaningful property at compile-time
+but has no runtime representation.
+
+For more details, see [SE-0302](https://github.com/apple/swift-evolution/blob/main/proposals/0302-concurrent-value-and-concurrent-closures.md#marker-protocols), which introduces marker protocols.
+
+At the moment, the language only has 1 marker protocol: `Sendable`.
+
+Fun fact: Rust has a very similar concept called
+[marker traits](https://doc.rust-lang.org/std/marker/index.html),
+including one called `Send`,
+which inspired the design of `Sendable`.
+
+## `@_nonEphemeral`
+
+Marks a function parameter that cannot accept a temporary pointer produced from
+an inout-to-pointer, array-to-pointer, or string-to-pointer conversion. Such a
+parameter may only accept a pointer that is guaranteed to outlive the duration
+of the function call.
+
+Attempting to pass a temporary pointer to an `@_nonEphemeral` parameter will
+produce a warning. This attribute is primarily used within the standard library
+on the various `UnsafePointer` initializers to warn users about
+the undefined behavior caused by using a temporary pointer conversion as an
+argument:
+
+```swift
+func baz() {
+  var x = 0
+
+  // warning: Initialization of 'UnsafePointer<Int>' results in a dangling pointer
+  let ptr = UnsafePointer(&x)
+
+  // warning: Initialization of 'UnsafePointer<Int>' results in a dangling pointer
+  let ptr2 = UnsafePointer([1, 2, 3])
+}
+```
+
+The temporary pointer conversion produces a pointer that is only
+guaranteed to be valid for the duration of the call to the initializer,
+and becomes invalid once the call ends.
+So the newly created `UnsafePointer` will be dangling.
+
+One exception to this is that inout-to-pointer conversions
+on static stored properties and global stored properties
+produce non-ephemeral pointers, as long as they have no observers:
+
+```swift
+var global = 0
+
+struct S {
+  static var staticVar = 0
+}
+
+func baz() {
+  let ptr = UnsafePointer(&global) // okay
+  let ptr2 = UnsafePointer(&S.staticVar) // okay
+}
+```
+
+Additionally, if they are of a tuple or struct type, their stored members
+without observers may also be passed inout as non-ephemeral pointers.
+
+For more details, see [the educational note on temporary pointer usage](/userdocs/diagnostics/temporary-pointers.md).
+
+## `@_nonoverride`
+
+## `@_objc_non_lazy_realization`
+
+## `@_optimize(...)`
```

Comment:
_optimize([none|size|speed]) controls the compiler's optimization mode. It is analagous to the command line options -Onone, -Osize, and -Ospeed, but limited to a single function body. `_optimizer(none)` can be used to diagnose compiler bugs or improve debugging in Release builds.